client/ts/graph_chart.ts

```diff
@@ -40,26 +40,57 @@ export function generateGraghChart(jsonUrl){
                 .append("circle")
                 .attr("r", nodeRadius)
                 // .style("opacity", "0.5");
-                .style('fill',  Color.White);
+                .style('fill',  Color.White)
+            .on("mouseover", nodeMouseOverBehavior)
+            .on("mouseout", nodeMouseOutBehavior);
 
         simulation.nodes(data.nodes)
             .on("tick", ticked);
         simulation.force("link").links(data.links);
 
         function ticked(){
             links
-                .attr("x1", function(d) { return d.source.x; })
-                .attr("y1", function(d) { return d.source.y; })
-                .attr("x2", function(d) { return d.target.x; })
-                .attr("y2", function(d) { return d.target.y; });
+                .attr("x1", function(d:any) { return d.source.x; })
+                .attr("y1", function(d:any) { return d.source.y; })
+                .attr("x2", function(d:any) { return d.target.x; })
+                .attr("y2", function(d:any) { return d.target.y; });
             nodes
-                .attr("cx", function(d) { return d.x; })
-                .attr("cy", function(d) { return d.y; })
+                .attr("cx", function(d:any) { return d.x; })
+                .attr("cy", function(d:any) { return d.y; })
                 .attr("stroke",  Color.White)
                 .attr("stroke-width", 1);
 
 
         }
     });
 
+}
+
+/**
+ * a function for mouse over behavior of the node in network graph. When user hovers over the node, display the feature
+ * name
+ * @param d: data, from the <circle> element's data attributes
+ * @param i: index
+ */
+function nodeMouseOverBehavior(d, i){
+    let svg = d3.select('svg');
+    svg.append("text")
+        .attr('id', d.source + '_index_' + d.index)
+        .attr('x', function() { return d.x - 50})
```

Comment:
maybe we'll improve the super-hard-coded values with design QA? :)

Reply:
yep!